crates/common/consensus/src/deneb/beacon_state.rs

```
@@ -823,8 +832,10 @@ impl BeaconState {
             let signing_root = compute_signing_root(deposit_message, domain);
             let sig = blst::min_pk::Signature::from_bytes(&signature.signature)
                 .map_err(|err| anyhow!("Failed to convert signiture type {err:?}"))?;
-            let public_key = PublicKey::from_bytes(&pubkey.inner)
-                .map_err(|err| anyhow!("Failed to convert pubkey type {err:?}"))?;
+            let public_key = match PublicKey::from_bytes(&pubkey.inner) {
+                Ok(pk) => pk,
+                Err(_) => return Ok(()), // Skip deposits with invalid public keys
```

I don't think this is right. It already ignores the corresponding deposit when pubkey is in wrong format. This just changes it to return Ok instead of proper error.

---

https://github.com/ReamLabs/ream/blob/5d5102886b22054a1a02455ceab9c84fc573d6c5/testing/ef-tests/src/macros.rs#L114-L129

Using `?` operator will propagate error, making this operation returns error. It made `key_validate_invalid_decompression` test case fail like: 

```
---- tests_process_deposit::test_operation stdout ----
Testing case: key_validate_invalid_decompression
thread 'tests_process_deposit::test_operation' panicked at testing/ef-tests/tests/tests.rs:69:1:
Test case key_validate_invalid_decompression should have succeeded but failed
```

I thought this should not be an error case.

Any feedback for this? Do I have to change the current macro(`test_operation`)?

---

Just check the expected post no? If they give an invalid key we want to propagate an error, we don't need to verify if the function ran successfully or not.

`key_validate_invalid_decompression` the test case is an error case, so it should error

---

https://github.com/ethereum/consensus-specs/tree/dev/tests/formats/operations#postssz_snappy

If `post.ssz_snappy` is present, the operation should not be aborted: `key_validate_invalid_decompression` has it 
 while other test castee like `invalid_wrong_deposit_for_deposit_count` doesn't.

AFAIK deposit contract doesn't validate the pubkey itself(it only validates the length). So user can put some invalid form in his deposit data, and that action shouldn't abort the state transition.

---

I've checked Prysm and Lighthouse implementations and I think @syjn99 is right. When `from_bytes` call to either of signature and public key fails, it should just return Ok(). Nice catch.

---

Good find @syjn99 :saluting_face: